File: baekjoon/회의실_배정.js
let [N, ...meetings] = require("fs")
  .readFileSync("/dev/stdin")
  .toString()
  .trim()
  .split("\n");
N = +N;
meetings = meetings.map((meeting) => meeting.split(" ").map(Number));

meetings.sort((a, b) => {
  if (a[1] === b[1]) return a[0] - b[0];
  return a[1] - b[1];
});

let cursor = 0;
let count = 0;

for (let i = 0; i < N; i++) {
  if (meetings[i][0] >= cursor) {
    cursor = meetings[i][1];
    count++;
  }
}

console.log(count);

File: baekjoon/회의실_배정.rs
use std::io::{BufReader,BufRead,stdin};

fn find_meeting_room(vector: Vec<(i32, i32)>) -> isize {
    let mut time = 0;
    let mut count = 0;
    for i in vector.into_iter() {
        if time <= i.1 {
            count += 1;
            time = i.0;
        }
    }
    count
}

fn main(){
    let mut reader = BufReader::new(stdin().lock());
    let mut input = String::new();
    reader.read_line(&mut input).unwrap();

    let n = input.trim().parse().unwrap();
    let mut v: Vec<(i32, i32)> = vec![];
    for _ in 0..n{
        input.clear();
        reader.read_line(&mut input).unwrap();
        let nums: Vec<i32> = input.trim().split_whitespace().map(|x|x.parse().unwrap()).collect::<Vec<i32>>();

        v.push((nums[1],nums[0]));
    }
    v.sort();

    let res = find_meeting_room(v);

    println!("{}", res);
} 